### PR TITLE
Add ability to search for all accounts when creating a list in web UI

### PR DESCRIPTION
**Pull request by Gargron**
You can only add accounts you follow to a list. However, since recently you can add follow-requested accounts to a list before they approve you. We can allow the user to search for all accounts when creating a list, and automatically follow those not already followed before adding them to the list.

image

Based on #32998.

### CODE REVIEW

File: app/javascript/mastodon/features/lists/members.tsx [hunk 1]
```diff
@@ -51,17 +55,51 @@ const AccountItem: React.FC<{
...
+              accountId,
+              onConfirm: () => {
+                apiFollowAccount(accountId)
+                  .then(() => apiAddAccountToList(listId, accountId))
+                  .then(() => {
+                    onToggle(accountId);
+                    return '';
+                  })
+                  .catch(() => {
+                    // Nothing
```
Comment (ClearlyClaire) [inline app/javascript/mastodon/features/lists/members.tsx:94]: I don't think errors should be ignored…

Comment (Gargron) [reply app/javascript/mastodon/features/lists/members.tsx:94]: On error, the UI will not mark the account as on the list, allowing the user to click "Follow" again. I don't have any designs for error states here, and not sure if alert notification in the corner would be helpful.

Comment (ClearlyClaire) [reply app/javascript/mastodon/features/lists/members.tsx:94]: I do think an alert notification in the corner would be somewhat helpful. We can certainly do better, but I think an error message is better than the UI silently not doing anything.

Comment (Gargron) [reply app/javascript/mastodon/features/lists/members.tsx]: Done!

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**